subprojects/core/src/main/java/org/gradle/execution/plan/TaskNodeFactory.java

```diff
@@ -83,7 +84,7 @@ public TaskNode getNode(Task task) {
     public TaskNode getOrCreateNode(Task task) {
         TaskNode node = nodes.get(task);
         if (node == null) {
-            if (task.getProject().getGradle() == thisBuild) {
+            if (((ProjectInternal) task.getProject()).getGradle().getIdentityPath().equals(thisBuild.getIdentityPath())) {
```

Comment:
That's interesting. I wonder how common it is to compare Gradle references to decide whether two projects are from the same build... 🤔